src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -209,38 +190,42 @@ private function createRelationsForItem($item, $formattedData)
      * Those will be nested accordingly in this relation array, so address relation will have a nested relation with country.
      *
      *
-     * @param  array  $input  The form data.
-     * @return array The formatted relation data.
+     * @param  array  $input  The form input.
+     * @return array The formatted relation details.
      */
-    private function getRelationDataFromFormData($data)
+    private function getRelationDetailsFromInput($input)
     {
-        $relation_fields = $this->getRelationFields();
+        $relation_fields = $this->getRelationFieldsWithoutPivot();
 
-        //remove fields that are not in the submitted form data
-        $relation_fields = array_filter($relation_fields, function ($item) use ($data) {
-            return Arr::has($data, $item['name']);
+        //remove fields that are not in the submitted form input
+        $relation_fields = array_filter($relation_fields, function ($item) use ($input) {
+            return Arr::has($input, $item['name']);
         });
 
-        $relationData = [];
+        $relationDetails = [];
         foreach ($relation_fields as $relation_field) {
+            // use the field name as a search parameter
             $attributeKey = $relation_field['name'];
 
-            if (! is_null(Arr::get($data, $attributeKey)) && isset($relation_field['pivot']) && $relation_field['pivot'] !== true) {
-                $key = implode('.relations.', explode('.', $this->getOnlyRelationEntity($relation_field)));
-                $fieldData = Arr::get($relationData, 'relations.'.$key, []);
-                if (! array_key_exists('model', $fieldData)) {
-                    $fieldData['model'] = $relation_field['model'];
-                }
-                if (! array_key_exists('parent', $fieldData)) {
-                    $fieldData['parent'] = $this->getRelationModel($attributeKey, -1);
-                }
-                $relatedAttribute = Arr::last(explode('.', $attributeKey));
-                $fieldData['values'][$relatedAttribute] = Arr::get($data, $attributeKey);
+            // we split the entity into relations, eg: user.accountDetails.address (user -> HasOne accountDetails -> BelongsTo address).
+            // we specifically use only the relation entity because relations like HasOne and MorphOne use the attribute in the relation string
+            $key = implode('.relations.', explode('.', $this->getOnlyRelationEntity($relation_field)));
 
-                Arr::set($relationData, 'relations.'.$key, $fieldData);
-            }
+            // since we can have for example 3 fields for address relation, we make sure that atleast once we set the relation details.
```

Comment:
```suggestion
            // since we can have for example 3 fields for address relation, we make sure that at least once we set the relation details.
```